checkstyle.xml

```diff
@@ -0,0 +1,393 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Just curious: Are these configurations handcrafted? If they are, should we consider using any predefined formats? e.g., `google-java-format` etc.,

Reply:
they are hand crafted, cassandra has a unique code style requirement: https://cassandra.apache.org/doc/latest/development/code_style.html

if we could use googles style id rather just run the style auto formatter on git commit so we just didnt have to think about it anymore but since we have to match a kinda 1-off thing this is best option i could find to maintain it.